pkg/util/inplaceupdate/inplace_update.go

```diff
@@ -161,6 +163,29 @@ func (c *realControl) Refresh(pod *v1.Pod, opts *UpdateOptions) RefreshResult {
 	return RefreshResult{RefreshErr: err}
 }
 
+func (c *realControl) SetPodNotReady(pod *v1.Pod) (bool, error) {
+	if isPodAlreadyNotReady(pod) {
```

Comment:
pod readiness status can be unstable,  pod can return to be ready but still waiting for deletion. Is it more safe to update the readiness condition even if the pod is not ready